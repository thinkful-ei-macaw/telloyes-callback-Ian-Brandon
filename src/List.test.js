import React from 'react';
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom';
import List from './List'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List
    header='header'
    cards={[{ id: 'a', title: 'First card', content: 'lorem ipsum' },
    { id: 'b', title: 'Second card', content: 'lorem ipsum' },
    { id: 'c', title: 'Third card', content: 'lorem ipsum' },
    { id: 'd', title: 'Fourth card', content: 'lorem ipsum' },]}
  />, div);
  ReactDOM.unmountComponentAtNode(div);
})

it('renders Cards as expected', () => {
  const tree = renderer
    .create(<List
      header='header'
      cards={[{ id: 'a', title: 'First card', content: 'lorem ipsum' },
      { id: 'b', title: 'Second card', content: 'lorem ipsum' },
      { id: 'c', title: 'Third card', content: 'lorem ipsum' },
      { id: 'd', title: 'Fourth card', content: 'lorem ipsum' },]}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});