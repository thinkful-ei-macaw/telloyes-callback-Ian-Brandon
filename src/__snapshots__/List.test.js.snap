// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Cards as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
